Add node groups

qemanuel opened the pull request:
## [4.0.0] - 2022-07-20

### Added
- AWS managed node groups
- multiple custom node groups
- NodeSelector input for loki and prometheus components
- Labels and taints for node groups

### Fixed
- [Loki issue #5909](https://github.com/grafana/loki/issues/5909#issuecomment-1120821579) with image v2.5.0-with-pr-6123-a630ae3

qedgardo commented:
Amazing work @qemanuel. You also replaced the ASG Launch configuration (deprecated) with the Launch Template.
You just could update the default value of the variable "k8s_auth_api" "client.authentication.k8s.io/v1alpha1" (deprecated) with "client.authentication.k8s.io/v1beta1"

qemanuel commented:
> Amazing work @qemanuel. You also replaced the ASG Launch configuration (deprecated) with the Launch Template. You just could update the default value of the variable "k8s_auth_api" "client.authentication.k8s.io/v1alpha1" (deprecated) with "client.authentication.k8s.io/v1beta1"

Sure, i can do that. It had that default because when we added that input [ PR #15 ](https://github.com/nimbux911/terraform-aws-eks/pull/15) , we didn't want to add a breaking change. But this PR have a lot of breaking changes, so i can update it

qemanuel commented:
> Amazing work @qemanuel. You also replaced the ASG Launch configuration (deprecated) with the Launch Template. You just could update the default value of the variable "k8s_auth_api" "client.authentication.k8s.io/v1alpha1" (deprecated) with "client.authentication.k8s.io/v1beta1"

Requested changes added
